compiler/ras/DebugCounter.cpp

```diff
@@ -413,8 +420,15 @@ TR::DebugCounter *TR::DebugCounterGroup::createCounter(const char *name, int8_t
       flags |= TR::DebugCounter::ContributesToDenominator;
       }
    TR::DebugCounter *result = new (persistentMemory) TR::DebugCounter(name, fidelity, denominator, flags);
+   TR_ASSERT(result, "DebugCounter *result must not be null. Ensure availability of persistent memory");
```

Comment:
In a build without asserts what will the behaviour be - should we just not add it to the list with some kind of trace warning or will other things fallover? How likely is this failure?